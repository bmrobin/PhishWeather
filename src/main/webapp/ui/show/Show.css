.show {
  margin-left: 20px;
}

.show-result-title {
  width: 100px;
}
