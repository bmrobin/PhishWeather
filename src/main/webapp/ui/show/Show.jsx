import React from 'react';
import './Show.css';

export class Show extends React.Component {

  constructor(props) {
    super(props);
  }
  
  markup() {
    return {
      __html: this.props.show.setlistnotes
    };
  }
  render() {

    return (
      <div className="show">
        <div>Phish show on <span id="show-date-id">{this.props.show.showdate}</span></div>

        <table>
          <tbody>
            <tr>
              <td className="show-result-title"><span><strong>Venue</strong></span></td>
              <td><span id="show-venue-id">{this.props.show.venue}</span></td>
            </tr>

            <tr>
              <td className="show-result-title"><span><strong>Location</strong></span></td>
              <td><span id="show-location-id">{this.props.show.location}</span></td>
            </tr>

            <tr>
              <td className="show-result-title"><span><strong>Show Notes</strong></span></td>
              <td><span id="show-notes-id" dangerouslySetInnerHTML={this.markup()}></span></td>
            </tr>

            <tr>
              <td className="show-result-title"><span><strong>phish.net Link</strong></span></td>
              <td><a href={this.props.show.link} target="blank">{this.props.show.showdate}</a></td>
            </tr>
          </tbody>
        </table>

      </div>
    );
  }
}
