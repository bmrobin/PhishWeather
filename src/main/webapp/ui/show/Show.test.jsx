import React from 'react';
import renderer from 'react-test-renderer';
import { Show } from './Show';

describe("Show component", () => {

  test("should render the show component", () => {
    let show = {
      showdate: '1998-08-09',
      venue: 'Virginia Beach Amphitheater',
      location: 'Virginia Beach, VA',
      setlistnotes: 'show notes here',
      link: 'http://phish.net'
    };
    let component = renderer.create(
      <Show show={show} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

});
