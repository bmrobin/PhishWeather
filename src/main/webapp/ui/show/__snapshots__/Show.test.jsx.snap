// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show component should render the show component 1`] = `
<div
  className="show"
>
  <div>
    Phish show on 
    <span
      id="show-date-id"
    >
      1998-08-09
    </span>
  </div>
  <table>
    <tbody>
      <tr>
        <td
          className="show-result-title"
        >
          <span>
            <strong>
              Venue
            </strong>
          </span>
        </td>
        <td>
          <span
            id="show-venue-id"
          >
            Virginia Beach Amphitheater
          </span>
        </td>
      </tr>
      <tr>
        <td
          className="show-result-title"
        >
          <span>
            <strong>
              Location
            </strong>
          </span>
        </td>
        <td>
          <span
            id="show-location-id"
          >
            Virginia Beach, VA
          </span>
        </td>
      </tr>
      <tr>
        <td
          className="show-result-title"
        >
          <span>
            <strong>
              Show Notes
            </strong>
          </span>
        </td>
        <td>
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "show notes here",
              }
            }
            id="show-notes-id"
          />
        </td>
      </tr>
      <tr>
        <td
          className="show-result-title"
        >
          <span>
            <strong>
              phish.net Link
            </strong>
          </span>
        </td>
        <td>
          <a
            href="http://phish.net"
            target="blank"
          >
            1998-08-09
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
